main.go

```diff
@@ -97,97 +128,21 @@ func main() {
 			http.Error(w, "error decoding request", http.StatusBadRequest)
 			return
 		}
-
-		log.Println("sent ids length: ", len(objList.ObjectIDs))
-		ready <- struct{}{}
+		jobs <- objList.ObjectIDs
 	})
 
-	resChan := make(chan result)
-	//goroutine that blocks until a new object id list has been received or context cancelled
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				errChan <- errors.New("context cancelled")
-				return
-
-			case <-ready:
-				log.Println("received ids length: ", len(objList.ObjectIDs))
-
-				for _, id := range objList.ObjectIDs {
-					go func(objectID int) {
-						details, err := fetchDetail(objectID)
-						if err != nil {
-							resChan <- result{err: err}
-						}
-
-						resChan <- result{objectDetail: details}
-					}(id)
-				}
-			}
-		}
-	}()
-
-	online := make([]ObjectDetail, 0)
-	offline := make([]ObjectDetail, 0)
+	for i := 0; i < req.workers; i++ {
```

Comment:
You messed up. You need only one filter worker, but several workers as in `worker` func. See detailed comment below (or above lol :D)